soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -3051,9 +3052,20 @@ void DrawRandoEditor(bool& open) {
     UIWidgets::Spacer(0);
 
     ImGui::Text("Seed");
-    if (ImGui::InputText("##RandomizerSeed", seedInputBuffer, MAX_SEED_BUFFER_SIZE, ImGuiInputTextFlags_CharsDecimal | ImGuiInputTextFlags_CallbackCharFilter, UIWidgets::TextFilters::FilterNumbers)) {
+    if (ImGui::InputText("##RandomizerSeed", seedInputBuffer, MAX_SEED_BUFFER_SIZE, ImGuiInputTextFlags_CharsNoBlank)) {
```

Comment:
If we want to allow string values to be entered, I wonder then if we'd want to increase the text field buffer size (currently just 10)

Numeric only strings should still correctly be bound to `uint32_t` range via `ImGui::DataTypeApplyFromText(seedInputBuffer, ImGuiDataType_U32, &seedInput, "%u");` below

Reply:
The point of the smaller buffer size was so that you couldn't input more characters than would fit in the data type, and thus the seed would always be EXACTLY what you typed in.  By allowing more characters, does this mean that people will be able to input a number-only string that won't fit in the data type, and then have it silently truncated so that the seed is no longer exactly what they typed in thanks to that function you mentioned?  If so, I very much don't like this.  

It shouldn't be hard to use callback to detect if the string is numbers-only as you're typing it, and then prevent more input beyond the old limit of 10 characters (the buffer was 11 so there was room for the null terminator), or if you paste in a number-only string that's longer than 10 characters, truncate it _immediately_ so that people see that number-only strings will be limited to what will fit in the data type.  (Or 9 characters if the 10-character string is still beyond the 2^32 range).

Reply:
The way it works in this PR (and prior to this PR), is any numeric only value beyond 2^32 is visually truncated when the input loses focus. For example, on the Gibbs-alfa right now, since the buffer is 10, when you enter 9999999999 you'll see it truncate.

Would you prefer it truncates as you type, rather than on defocus? Or maybe we could add a visual indicator the the field will truncate when defocus (like add a border color to the input if we can)

Reply:
Ideally it should happen as you type since it's possible for the focus loss to happen BECAUSE somebody presses the Generate button, which means that they'd be getting a different seed than they thought they input.  That would also make it possible to fix the case I mentioned above where you're within the 10 char limit but beyond the 2^32 limit and thus will only be able to keep 9 characters.  

In fact, I'd go so far as to say that the box should NEVER be able to contain more characters than will fit in the buffer, meaning that we'd have to do truncation as you type, but I'd also be just as happy with some sort of visual indicator like a border color to indicate that truncation will _eventually_ happen as long as the indicator itself shows up immediately as soon as we detect that it will be necessary.  

The fact that we currently _don't_ do this and are thus possibly surprising people by truncating their seeds is admittedly a bug that's not new to this PR though, so I will also grudgingly accept if instead a new issue is opened so we can fix it soon after this PR is complete.